### ISSUE TITLE
add more Language support for vita3k gui

### ISSUE THREAD

**Issue by ikhoeyZX** [Vita3K/Vita3K#3071]
we know that someone already make vita3k other language version in illegal way, so can we request make vita3k gui support more language (even it not supported by vita it self)?

**Comment by Macdu** [comment]
If you want to add support for an additional language in Vita3K, you can make a pull request for it.

**Comment by ikhoeyZX** [comment]
how?

**Comment by Croden1999** [comment]
my suggestion is need add console language before add system language, same as likes to rpcs3, yuzu etc.

**Comment by Macdu** [comment]
Yes I agree, that sounds like the most logical idea.

**Comment by Kitakatarashima** [comment]
how to add console language??

**Comment by Zangetsu38** [comment]
> my suggestion is need add console language (for games) before add system language (for gui), same as likes to rpcs3, yuzu etc.

rpcs3 don't have this, rpcs3 have ONLY like viota3k for System language, and gui is only in english
![image](https://github.com/Vita3K/Vita3K/assets/5261759/77b202be-641f-4c17-ab03-4754f45e9196)

**Comment by Croden1999** [comment]
> > my suggestion is need add console language (for games) before add system language (for gui), same as likes to rpcs3, yuzu etc.
> 
> rpcs3 does not have this, rpcs3 only has console language, like vita3k, also called system language, and the GUI is only in English. ![image](https://user-images.githubusercontent.com/5261759/283318417-77b202be-641f-4c17-ab03-4754f45e9196.png)
> 
> > how to add console language??
> 
> This is just not possible with the current code if this language does not exist on the psvita

but rpcs3 support custom lang.qt/ts files right?

**Comment by Zangetsu38** [comment]
> but rpcs3 support custom translations .qt/ts files right?

I have never see that, this option is literally April fool lol
![image](https://github.com/Vita3K/Vita3K/assets/5261759/de4c6f2b-ca01-44dc-bcb0-f06576383641)

**Comment by Croden1999** [comment]
> > but rpcs3 support custom translations .qt/ts files right?
> 
> I have never see that, this option is literally April fool lol, english is hard coded ![image](https://user-images.githubusercontent.com/5261759/283325368-de4c6f2b-ca01-44dc-bcb0-f06576383641.png)

i knew only english have in master, the users can pust the translation files in qt6\translations folder

![image](https://github.com/Vita3K/Vita3K/assets/61804715/a7e00f11-216e-4901-a923-a774eb74ff14)
![image](https://github.com/Vita3K/Vita3K/assets/61804715/2292cc51-bb71-4d45-9e2d-c48c40abc134)
![image](https://github.com/Vita3K/Vita3K/assets/61804715/15eeac3b-4d1c-4b5d-8fc6-3e0448fbd88c)

**Comment by Zangetsu38** [comment]
> > > but rpcs3 support custom translations .qt/ts files right?
> > 
> > 
> > I have never see that, this option is literally April fool lol, english is hard coded ![image](https://user-images.githubusercontent.com/5261759/283325368-de4c6f2b-ca01-44dc-bcb0-f06576383641.png)
> 
> I know that in the master version there is only English, the users can pust the translation files in qt6\translations folder
> 
> ![image](https://user-images.githubusercontent.com/61804715/283327230-a7e00f11-216e-4901-a923-a774eb74ff14.png) ![image](https://user-images.githubusercontent.com/61804715/283327308-2292cc51-bb71-4d45-9e2d-c48c40abc134.png) ![image](https://user-images.githubusercontent.com/61804715/283327342-15eeac3b-4d1c-4b5d-8fc6-3e0448fbd88c.png)

Oh, i see 🤔, initial adding this option is litteraly big troll lol
I have no know have adding task of this later

**Comment by Kitakatarashima** [comment]
maybe you can but Vita3k has to have an external language instead of the original PS Vita and create your own folder, that's what I think, sorry, I still don't really understand

**Comment by Kitakatarashima** [comment]
If you can add another language besides the language in Vita3K, I'm ready to help translate Indonesian

**Comment by Zangetsu38** [comment]
> If you can add another language besides the language in Vita3K, I'm ready to help translate Indonesian

you can't just add a language like that, that's not how it works
for rpcs3, it doesn't have any language included, it's quite différent.

**Comment by Croden1999** [comment]
> > If you can add another language besides the language in Vita3K, I'm ready to help translate Indonesian
> 
> you can't just add a language like that, that's not how it works for rpcs3, it doesn't have any language included, it's quite différent.

we should could add custom language support, but keep en console language like that.

**Comment by Zangetsu38** [comment]
> we should could add custom translation support, but keep sys_lang en console language like that, even the language no exist on psvita.

I just explained that it wasn't possible, rpcs3 doesn't have any language included, it's totally different.

rpcs3 is all custom language so...

the only solution is to create another language enum for the interface languages... except that it's totally stupid..

it makes no sense to rewrite everything, to add a system/gui separation for random languages ​​which I repeat does not exist anywhere

and in this case, especially for a single language..

**Comment by Kitakatarashima** [comment]
I really don't understand coding, I just want to help translate, I don't really know anything, but I can just translate, that's what I said, maybe I can, I can't give you advice, this or this or not, I don't know what's what, that's what, I can't help more, I just Can you help me as much as I can? Yes, I know it's not that easy to write a language, but I'm just saying I'll help translate so that the Indonesian language is pleasant to hear, not just from Google Translate, you know?

**Comment by Macdu** [comment]
I think the option to add a custom language support would be pretty straightforward to add (just a checkbox "Use a custom language for the GUI" or something like that), shouldn't require a lot of changes.



> it makes no sense to rewrite everything, to add a system/gui separation for random languages ​​which I repeat does not exist anywhere

Yeah, you're right, Indonesian is a random language for a random country which is absolutely not the country with the most users in the discord server. I can't see any reason anyone would be interested by having an Indonesian translation of the UI.
![image](https://github.com/Vita3K/Vita3K/assets/5671744/68af8c54-5003-44eb-b8d1-ef0d388b554d)

**Comment by Kitakatarashima** [comment]
Who said that I want to make my own language files, the language files come from you guys, I can't code programs, that's me, I just want to help translate, I said, if you can add another language, if you can't, that's okay, that's okay

**Comment by Macdu** [comment]
I'll (or Croden if he wants) add an option to load a custom translation file for the GUI.

**Comment by Macdu** [comment]
The best option would be to separate the GUI language and the game language but that would require a bit more work, if anyone is up for it.

**Comment by Croden1999** [comment]
> I'll (or Croden if he wants) add an option to load a custom translation file for the GUI.

the users need it, not for me, sir. 😂

**Comment by Macdu** [comment]
> > I'll (or Croden if he wants) add an option to load a custom translation file for the GUI.
> 
> the users need it, not for me, sir. 😂 i no have problems.

I meant if you want to implement it, otherwise I'll do it.

**Comment by Croden1999** [comment]
> I meant if you want to implement it, otherwise I'll do it.


you implement it, sir, your code better

**Comment by Zangetsu38** [comment]
> I think the option to add a custom language support would be pretty straightforward to add (just a checkbox "Use a custom language for the GUI" or something like that), shouldn't require a lot of changes.

you have to rewrite everything, and it would be a huge mess for 0 interest...

this language does not exist in the psvita I remind you, so there is neither application system, nor any games which use it...

So the only question is why do all this if everything else will be only in English?
Live area/trophy/Manuel keep in english...like game...

for the option to load a custom language, it's a little more complicated than that anyway...

> Yeah, you're right, Indonesian is a random language for a random country which is absolutely not the country with the most users in the discord server. I can't see any reason anyone would be interested by having an Indonesian translation of the UI. ![image](https://user-images.githubusercontent.com/5671744/283417302-68af8c54-5003-44eb-b8d1-ef0d388b554d.png)

Yes it's a random language, so what?  
Have you seen this language on a single PlayStation console?  
No,  Why ?

Is not my fault if this langue no exist anywhere.
And show % of discord have no sense
Do you also plan to add and update all the languages ​​that also exist?

I remind you, that I coded language support ONLY for existing languages ​​in the console, and basically only for system applications, using the official language file for everything.

**Comment by Zangetsu38** [comment]
> Who said that I want to make my own language files, the language files come from you guys, I can't code programs, that's me, I just want to help translate, I said, if you can add another language, if you can't, that's okay, that's okay

Problem here is not translation, is this is the benefit that additional work can have, both in addition and in maintenance.

**Comment by nishinji** [comment]
> > I think the option to add a custom language support would be pretty straightforward to add (just a checkbox "Use a custom language for the GUI" or something like that), shouldn't require a lot of changes.
> 
> you have to rewrite everything, and it would be a huge mess for 0 interest...
> 
> this language does not exist in the psvita I remind you, so there is neither application system, nor any games which use it...
> 
> So the only question is why do all this if everything else will be only in English? Live area/trophy/Manuel keep in english...like game...
> 
> for the option to load a custom language, it's a little more complicated than that anyway...
> 
> > Yeah, you're right, Indonesian is a random language for a random country which is absolutely not the country with the most users in the discord server. I can't see any reason anyone would be interested by having an Indonesian translation of the UI. ![image](https://user-images.githubusercontent.com/5671744/283417302-68af8c54-5003-44eb-b8d1-ef0d388b554d.png)
> 
> Yes it's a random language, so what? Have you seen this language on a single PlayStation console? No, Why ?
> 
> Is not my fault if this langue no exist anywhere. And show % of discord have no sense Do you also plan to add and update all the languages ​​that also exist?
> 
> I remind you, that I coded language support ONLY for existing languages ​​in the console, and basically only for system applications, using the official language file for everything.

That attitude of yours is like the APPLE you hate ...
Don't try to unnecessarily restrict what is of benefit to users.
we can add translation for UI, not for system.

**Comment by Kitakatarashima** [comment]
Yes, if that's the case, I agree, but we don't need the full Besic language, it's still English, it doesn't matter, what's important is the language, the language that's important for Vita3K can be changed to another language other than the one in Vita3K. In my opinion, the change is just gradual. Like other emulators, it doesn't immediately fill in the language. Later, I'm sure there will be a way to change the system language if that's what I think

**Comment by Zangetsu38** [comment]
> That attitude of yours is like the APPLE you hate ... Don't try to unnecessarily restrict what is of benefit to users. we can add translation for UI, not for system.

No, i just explain fact, In case you don't know, the user interface goes hand in hand with the language, system, just look at the code to understand, the 2 are linked and work together everywhere.
YOU CAN'T CHANGE IT...

The user interface that I coded IS the psvita app system
everything else added after that was a bonus that didn't have an official file for the language.

I coded this only to translate what was already available on the psvita itself, and therefore used the official translation.

Precisely to have limited language files with fixed/static data which would NEVER change, (well except in case of error detection), precisely because having fun updating 19 Language in real time with each change, it's is a real mess...

Not for nothing that I didn't want it in the setting dialog
It wasn't to piss off, it wasn't a quick maintenance measure..
Devs who add options don't have to waste time translating each option into EACH language...
And since it evolves constantly, long live the shit

Otherwise, for this idea, what Macdu proposes on the other hand is feasible, but requires a bit of refactoring.
So by creating an option in the GUI example tab, call example: "Enable custom language" and can select with open file, or with a list of custom detect language files, inside "lang/custom" perhaps, so that they can then open it and search for the custom language, which is what rpcs3 does in the end.

> Yes, if that's the case, I agree, but we don't need the full Besic language, it's still English, it doesn't matter, what's important is the language, the language that's important for Vita3K can be changed to another language other than the one in Vita3K. In my opinion, the change is just gradual. Like other emulators, it doesn't immediately fill in the language. Later, I'm sure there will be a way to change the system language if that's what I think

So, no, it will never be possible to change the system language, dsl...
If he is not in the psvita, it will never exist...

**Comment by Macdu** [comment]
Don't worry, pay no attention to the rant above. Changing the system language to one not supported by the PS Vita OS will be supported at some point not too long from now.

**Comment by Zangetsu38** [comment]
> Don't worry, pay no attention to the rant above. Changing the system language to one not supported by the PS Vita OS will be supported at some point not too long from now.

Read me instead of answering that, he was talking about the system language of the psvita, which works with the games and the rest, it's absolutely impossible, the system language is fixed, you can't add another, it's is like that and that's it, don't confuse him anymore...

I explained how to add custom languages to the interface, but it will forever remain limited to the interface...

**Comment by Croden1999** [comment]
Could we close this issue? It is implemented.

**Comment by Zangetsu38** [comment]
Yes but i cant

**Comment by Macdu** [comment]
It's does not entirely fix this issue. The end term goal would be adding these languages directly to Vita3K instead of having a user have to put the lang file in the appropriate folder.
